File: leetcode/0001_two_sum.py
# https: // leetcode.com/problems/two-sum/


from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indices_map = dict()
        for index, value in enumerate(nums):
            if (target - value) in indices_map:
                return [indices_map[(target - value)], index]
            indices_map[value] = index

File: leetcode_go/0001-two-sum.go
//https://leetcode.com/problems/two-sum/
package leetcodego

func twoSum(nums []int, target int) []int {
	dict := make(map[int]int)
	for index, num := range nums {
		i, found := dict[num]
		if found {
			return []int{i, index}
		}
		dict[target-num] = index
	}
	return []int{}
}
